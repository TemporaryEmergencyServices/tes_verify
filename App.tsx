import { StatusBar } from 'expo-status-bar';
import React from 'react';
import { SafeAreaProvider } from 'react-native-safe-area-context';

import useCachedResources from './hooks/useCachedResources';
import useColorScheme from './hooks/useColorScheme';
import Navigation from './navigation';


//The following are for redux
import {createStore, applyMiddleware} from 'redux'
import {Provider} from 'react-redux'
import thunkMiddleware from 'redux-thunk'
// import reducer from './reducers'
import rootReducer from './reducers'

const middleware = applyMiddleware(thunkMiddleware)
export const store = createStore(rootReducer, middleware)
// const store = createStore(rootReducer)
//End redux

export default function App() {
  const isLoadingComplete = useCachedResources();
  const colorScheme = useColorScheme();

  if (!isLoadingComplete) {
    return null;
  } else {
    return (
      //Provider wrapper and store area for redux
      <Provider store={store}>
        <SafeAreaProvider>
            <Navigation colorScheme={colorScheme} />
            <StatusBar />
        </SafeAreaProvider>
      </Provider>
    );
  }
}