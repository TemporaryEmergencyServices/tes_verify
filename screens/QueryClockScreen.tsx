import firebase from '../firebase.js'
import '@firebase/firestore'
import React, { useState } from 'react';

import { useSelector, RootStateOrAny } from 'react-redux'
import { StyleSheet, Text, View, TextInput, TouchableOpacity, Alert, Button } from 'react-native';

import { useDispatch } from 'react-redux'
import { ScrollView } from 'react-native-gesture-handler';
import DatePicker from 'react-native-datepicker';
//date picker code from https://snack.expo.io/@aboutreact/date-picker-example?session_id=snack-session-TT5a8E!7q

export default function QueryClockScreen({ navigation }) {
  const today = new Date()
  const [startDateState,setStartDateState] = useState(today.getFullYear().toString() + '-' + (today.getMonth() + 1).toString().padStart(2, '0') + '-' + today.getDate().toString().padStart(2, '0'))
  const [stopDateState,setStopDateState] = useState(today.getFullYear().toString() + '-' + (today.getMonth() + 1).toString().padStart(2, '0') + '-' + today.getDate().toString().padStart(2, '0'))
  const [userIdState,setUserIdState] = useState('')
  const [ethnicityState, setEthnicityState] = useState('')
  const [sexState, setSexState] = useState('')
  const [firstNameState, setFirstNameState] = useState('')
  const [lastNameState, setLastNameState] = useState('')


  const handleSearch = () => {
    navigation.push('DisplayQueryClockScreen', {
        firstName: firstNameState,
        lastName: lastNameState,
        userId: userIdState.toLowerCase(),
        ethnicity: ethnicityState,
        sex: sexState,
        startDate: startDateState,
        stopDate: stopDateState
    })
  }

  return (
    <View style={styles.container}>
      <Text style={styles.instructions}>Search for records by name, userid, gender, ethnicity, and/or date.</Text>
      <View style={styles.inputView} >
        <TextInput  
          style={styles.inputText}
          placeholder="First Name" 
          placeholderTextColor="grey"
          onChangeText={text => setFirstNameState(text)}/>
      </View>

      <View style={styles.inputView} >
        <TextInput  
          style={styles.inputText}
          placeholder="Last Name" 
          placeholderTextColor="grey"
          onChangeText={text => setLastNameState(text)}/>
      </View>

      <View style={styles.inputView} >
        <TextInput  
          style={styles.inputText}
          placeholder="Volunteer Email" 
          placeholderTextColor="grey"
          onChangeText={text => setUserIdState(text)}/>
      </View>

      <View style={styles.inputView} >
        <TextInput  
          style={styles.inputText}
          placeholder="Ethnicity" 
          placeholderTextColor="grey"
          onChangeText={text => setEthnicityState(text)}/>
      </View>

      <View style={styles.inputView} >
        <TextInput  
          style={styles.inputText}
          placeholder="Gender (M/F)" 
          placeholderTextColor="grey"
          onChangeText={text => setSexState(text)}/>
      </View>
      <Text style={styles.dateRange}> Date Range (from, to): </Text>
      <DatePicker
          style={styles.datePickerStyle}
          date = {startDateState}
          mode="date" //The enum of date, datetime and time
          placeholder={startDateState}
          format="YYYY-MM-DD"
          confirmBtnText="Confirm"
          cancelBtnText="Cancel"
          customStyles={{
            dateIcon: {
              //display: 'none',
              position: 'absolute',
              left: 0,
              top: 4,
              marginLeft: 0,
            },
            dateInput: {
              marginLeft: 36,
            },
          }}
          onDateChange={(date) => {
            setStartDateState(date);
          }}
        />
       
        <DatePicker
          style={styles.datePickerStyle}
          date = {stopDateState}
          mode="date" //The enum of date, datetime and time
          placeholder={stopDateState}
          format="YYYY-MM-DD"
          confirmBtnText="Confirm"
          cancelBtnText="Cancel"
          customStyles={{
            dateIcon: {
              //display: 'none',
              position: 'absolute',
              left: 0,
              top: 4,
              marginLeft: 0,
            },
            dateInput: {
              marginLeft: 36,
            },
          }}
          onDateChange={(date) => {
            setStopDateState(date);
          }}
        />
      
      <TouchableOpacity style={styles.loginBtn} onPress={handleSearch}>
        <Text style={styles.signUpText} >SEARCH</Text>
      </TouchableOpacity>

      <Button 
        title="LEAVE PAGE" 
        color = "#1C5A7D" 
        onPress={() => navigation.goBack()} />

    </View>
  );
}

const styles = StyleSheet.create({
    scrollview: {
      fontSize: 20,
      fontWeight: 'bold',
      justifyContent: 'center',
      textAlign: 'center',
      paddingTop: 60,
      paddingBottom: 15,
      color: "#1C5A7D",
      
    },
    container: {
      flex: 1,
      backgroundColor: '#FFFFFF',
      alignItems: 'center',
      justifyContent: 'center',
    },
    contentContainer: {
      backgroundColor: '#FFFFFF',
      alignItems: 'center',
      justifyContent: 'center',
      width: 300
    },
  
    logo:{
      fontWeight:"bold",
      fontSize:24,
      color:"#1C5A7D",
      textAlign: 'center',
      marginTop: 60,
      paddingRight: 30, 
      paddingLeft: 30
    },
  
    instructions:{
      fontSize:18,
      color:"#1C5A7D",
      marginBottom:40,
      textAlign: 'center',
      paddingRight: 7,
      paddingLeft: 7
    },
    dateRange:{
      fontSize:16,
      fontWeight: 'bold',
      color:"black",
      marginBottom:7,
      textAlign: 'center',
      paddingRight: 7,
      paddingLeft: 7
    },
    inputView:{
      width:270,
      backgroundColor:"#2B2E32",
      borderRadius:25,
      height:50,
      marginBottom:20,
      justifyContent:"center",
      padding: 20,
      paddingRight: 40,
      
    },
    inputText:{
      height:50,
      color:"white"
    },
    forgot:{
      color:"white",
      fontSize:11
    },
    loginBtn:{
      width:"80%",
      backgroundColor:"#E11383",
      borderRadius:25,
      height:50,
      alignItems:"center",
      justifyContent:"center",
      marginTop: 20,
      marginBottom: 5
    },
    createAccountBtn:{
      width:"80%",
      backgroundColor:"#E11383",
      borderRadius:25,
      height:50,
      alignItems:"center",
      justifyContent:"center",
      marginTop:10,
      marginBottom:10
    },
    returnToSignInText:{
      color:"black"
    },
    signUpText:{
      color:"white",
      fontSize: 18,
      fontWeight: "bold", 
    },
    input: {
      margin: 5,
      padding: 6,
      borderRadius: 8,
      marginBottom: 8,
      paddingHorizontal: 10,
      backgroundColor: "#eceff1",
    },
    datePickerStyle: {
        width: 200,
        marginTop: 7,
        marginBottom: 7,
      },
  })
  
  const modalstyles = StyleSheet.create({
    centeredView: {
      flex: 1,
      justifyContent: "center",
      alignItems: "center",
      marginTop: 22
    },
    modalView: {
      margin: 20,
      backgroundColor: "white",
      borderRadius: 20,
      padding: 35,
      alignItems: "center",
      shadowColor: "#000",
      shadowOffset: {
        width: 0,
        height: 2
      },
      shadowOpacity: 0.25,
      shadowRadius: 3.84,
      elevation: 5
    },
    openButton: {
      backgroundColor: "#F194FF",
      borderRadius: 20,
      padding: 10,
      elevation: 2
    },
    textStyle: {
      color: "black",
      fontWeight: "bold",
      textAlign: "left",
      fontSize: 18
    },
    modalText: {
      marginBottom: 15,
      textAlign: "center"
    }
  });