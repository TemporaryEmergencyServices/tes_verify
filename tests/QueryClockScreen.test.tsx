import React from "react";
import renderer from 'react-test-renderer';
import { act, fireEvent, render } from '@testing-library/react-native';
import QueryClockScreen from "../screens/QueryClockScreen"
import { Provider } from 'react-redux'

import { store } from '../App'

jest.useFakeTimers()

describe("<QueryClockScreen />", () => {
    const navigation={}
    const tree = renderer.create(<Provider store={store} ><QueryClockScreen navigation={navigation}/></Provider>)
    const wrapper = tree.toJSON()

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    });

    it('renders unauthorized view', () => {
        const text = wrapper.children[0].children[0]
        expect(text).toBe('Search for records by name, userid, gender, ethnicity, and/or date.')
    })
});