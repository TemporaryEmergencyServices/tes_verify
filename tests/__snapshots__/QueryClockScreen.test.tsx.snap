// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QueryClockScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#1C5A7D",
        "fontSize": 18,
        "marginBottom": 40,
        "paddingLeft": 7,
        "paddingRight": 7,
        "textAlign": "center",
      }
    }
  >
    Search for records by name, userid, gender, ethnicity, and/or date.
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#2B2E32",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 20,
        "padding": 20,
        "paddingRight": 40,
        "width": 270,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="First Name"
      placeholderTextColor="grey"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "white",
          "height": 50,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#2B2E32",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 20,
        "padding": 20,
        "paddingRight": 40,
        "width": 270,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Last Name"
      placeholderTextColor="grey"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "white",
          "height": 50,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#2B2E32",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 20,
        "padding": 20,
        "paddingRight": 40,
        "width": 270,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Volunteer Email"
      placeholderTextColor="grey"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "white",
          "height": 50,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#2B2E32",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 20,
        "padding": 20,
        "paddingRight": 40,
        "width": 270,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Ethnicity"
      placeholderTextColor="grey"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "white",
          "height": 50,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#2B2E32",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 20,
        "padding": 20,
        "paddingRight": 40,
        "width": 270,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Gender (M/F)"
      placeholderTextColor="grey"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "white",
          "height": 50,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 7,
        "paddingLeft": 7,
        "paddingRight": 7,
        "textAlign": "center",
      }
    }
  >
     Date Range (from, to): 
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "width": 142,
        },
        Object {
          "marginBottom": 7,
          "marginTop": 7,
          "width": 200,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderColor": "#aaa",
              "borderWidth": 1,
              "flex": 1,
              "height": 40,
              "justifyContent": "center",
            },
            Object {
              "marginLeft": 36,
            },
            false,
            false,
          ]
        }
      >
        <Text
          allowFontScaling={true}
          style={
            Array [
              Object {
                "color": "#333",
              },
              undefined,
            ]
          }
        >
          2020-12-02
        </Text>
      </View>
      <Image
        source={1}
        style={
          Array [
            Object {
              "height": 32,
              "marginLeft": 5,
              "marginRight": 5,
              "width": 32,
            },
            Object {
              "left": 0,
              "marginLeft": 0,
              "position": "absolute",
              "top": 4,
            },
          ]
        }
      />
      <Modal
        animationType="none"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        supportedOrientations={
          Array [
            "portrait",
            "portrait-upside-down",
            "landscape",
            "landscape-left",
            "landscape-right",
          ]
        }
        transparent={true}
        visible={false}
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "flex-end",
                "backgroundColor": "#00000077",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "height": 0,
                    "overflow": "hidden",
                  }
                }
              >
                <View
                  pointerEvents="auto"
                >
                  <View
                    style={
                      Array [
                        Object {
                          "borderTopColor": "#ccc",
                          "borderTopWidth": 1,
                          "marginTop": 42,
                        },
                        undefined,
                      ]
                    }
                  >
                    <RCTDatePicker
                      date={1606888800000}
                      mode="date"
                      onChange={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "height": 216,
                        }
                      }
                      timeZoneOffsetInMinutes={null}
                    />
                  </View>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 42,
                        "justifyContent": "center",
                        "paddingHorizontal": 20,
                        "position": "absolute",
                        "top": 0,
                      },
                      Object {
                        "left": 0,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    allowFontScaling={true}
                    style={
                      Array [
                        Object {
                          "color": "#46cf98",
                          "fontSize": 16,
                        },
                        Object {
                          "color": "#666",
                        },
                        undefined,
                      ]
                    }
                  >
                    Cancel
                  </Text>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 42,
                        "justifyContent": "center",
                        "paddingHorizontal": 20,
                        "position": "absolute",
                        "top": 0,
                      },
                      Object {
                        "right": 0,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    allowFontScaling={true}
                    style={
                      Array [
                        Object {
                          "color": "#46cf98",
                          "fontSize": 16,
                        },
                        undefined,
                      ]
                    }
                  >
                    Confirm
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "width": 142,
        },
        Object {
          "marginBottom": 7,
          "marginTop": 7,
          "width": 200,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderColor": "#aaa",
              "borderWidth": 1,
              "flex": 1,
              "height": 40,
              "justifyContent": "center",
            },
            Object {
              "marginLeft": 36,
            },
            false,
            false,
          ]
        }
      >
        <Text
          allowFontScaling={true}
          style={
            Array [
              Object {
                "color": "#333",
              },
              undefined,
            ]
          }
        >
          2020-12-02
        </Text>
      </View>
      <Image
        source={1}
        style={
          Array [
            Object {
              "height": 32,
              "marginLeft": 5,
              "marginRight": 5,
              "width": 32,
            },
            Object {
              "left": 0,
              "marginLeft": 0,
              "position": "absolute",
              "top": 4,
            },
          ]
        }
      />
      <Modal
        animationType="none"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        supportedOrientations={
          Array [
            "portrait",
            "portrait-upside-down",
            "landscape",
            "landscape-left",
            "landscape-right",
          ]
        }
        transparent={true}
        visible={false}
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "flex-end",
                "backgroundColor": "#00000077",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "height": 0,
                    "overflow": "hidden",
                  }
                }
              >
                <View
                  pointerEvents="auto"
                >
                  <View
                    style={
                      Array [
                        Object {
                          "borderTopColor": "#ccc",
                          "borderTopWidth": 1,
                          "marginTop": 42,
                        },
                        undefined,
                      ]
                    }
                  >
                    <RCTDatePicker
                      date={1606888800000}
                      mode="date"
                      onChange={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "height": 216,
                        }
                      }
                      timeZoneOffsetInMinutes={null}
                    />
                  </View>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 42,
                        "justifyContent": "center",
                        "paddingHorizontal": 20,
                        "position": "absolute",
                        "top": 0,
                      },
                      Object {
                        "left": 0,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    allowFontScaling={true}
                    style={
                      Array [
                        Object {
                          "color": "#46cf98",
                          "fontSize": 16,
                        },
                        Object {
                          "color": "#666",
                        },
                        undefined,
                      ]
                    }
                  >
                    Cancel
                  </Text>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 42,
                        "justifyContent": "center",
                        "paddingHorizontal": 20,
                        "position": "absolute",
                        "top": 0,
                      },
                      Object {
                        "right": 0,
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    allowFontScaling={true}
                    style={
                      Array [
                        Object {
                          "color": "#46cf98",
                          "fontSize": 16,
                        },
                        undefined,
                      ]
                    }
                  >
                    Confirm
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#E11383",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 5,
        "marginTop": 20,
        "opacity": 1,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      SEARCH
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#1C5A7D",
            },
          ]
        }
      >
        LEAVE PAGE
      </Text>
    </View>
  </View>
</View>
`;
